tests/supervisors/test_watchgodreload.py

```diff
@@ -13,27 +15,30 @@ def run(sockets):
 
 def test_statreload(certfile_and_keyfile):
     config = Config(app=None)
-    reloader = WatchdogReload(config, target=run, sockets=[])
+    reloader = WatchGodReload(config, target=run, sockets=[])
     reloader.signal_handler(sig=signal.SIGINT, frame=None)
     reloader.run()
 
 
-def test_should_reload(tmpdir):
-    update_file = Path(os.path.join(str(tmpdir), "example.py"))
+@pytest.mark.parametrize(
+    "should_reload, file", [(True, "example.py"), (False, ".dotted")]
+)
+def test_should_reload(tmpdir, should_reload, file):
```

Comment:
Please name scenarios which parameters test, or better, create two separate tests:

- `test_should_reload_is_true_when_python_file_is_changed`
- `test_should_reload_is_false_when_dot_file_is_changed`

Parametrization obfuscates the intention between assertions and leads to bad growth when more parametrized scenarios are bolted on on top of old parameters set, further hiding the intention behind tests.